src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1132,6 +1173,18 @@ public Void call() throws Exception {
                 }
             }, new NoRetryStrategy());
         }
+
+    }
+
+    public static void removeImage(final Azure azureClient, final String vmName, final String resourceGroupName) {
+        List<VirtualMachineCustomImage> customImages = azureClient.virtualMachineCustomImages().listByGroup(resourceGroupName);
+        for (VirtualMachineCustomImage image : customImages) {
+            String prefix = StringUtils.substringBefore(image.name(), "Image");
+            if (StringUtils.contains(vmName, prefix)) {
```

Comment:
Agree with Ken, why not pass the name of the image to be deleted as parameter? Are we able to know exactly its name?